synapse/config/server.py

```diff
@@ -184,49 +184,74 @@ def generate_ip_set(
 
 @attr.s(frozen=True)
 class HttpResourceConfig:
-    names = attr.ib(
-        type=List[str],
+    names: List[str] = attr.ib(
         factory=list,
         validator=attr.validators.deep_iterable(attr.validators.in_(KNOWN_RESOURCES)),  # type: ignore
     )
-    compress = attr.ib(
-        type=bool,
+    compress: bool = attr.ib(
         default=False,
         validator=attr.validators.optional(attr.validators.instance_of(bool)),  # type: ignore[arg-type]
```

Comment:
Is this a style preference or a functional change? (Perhaps mypy et al only recognise the annotation?)

Reply:
This is a style preference, mypy pulls it out either way.

If `auto_attribs=True` is used, it is required.